torque-core/src/main/kotlin/com/workday/torque/TestChunkRunner.kt

```diff
@@ -17,32 +19,49 @@ class TestChunkRunner(
         return try {
             adbDevice.log("Starting tests...")
             installer.ensureTestPackageInstalled(args, testChunk)
+            makeTestFileDirectories(args).await()
             runAndParseTests(args.chunkTimeoutSeconds, testChunk).await()
         } catch (e: Exception) {
             adbDevice.log("TestChunk run crashed with exception: ${e.message}")
             createCrashedAdbDeviceTestResults(testChunk, e)
         }
     }
 
+    private fun makeTestFileDirectories(args: Args): Completable {
+        return processRunner.runAdb(commandAndArgs = listOf(
+                        "-s", adbDevice.id,
+                        "shell", "mkdir -p ${args.testFilesPullDeviceDirectory}/coverage-reports"
+                ),
+                        destroyOnUnsubscribe = true)
+                .ofType(Notification.Exit::class.java)
+                .map { true }
+                .doOnError { error -> adbDevice.log("Failed to mkdir on ${adbDevice.tag}, filepath: ${args.testFilesPullDeviceDirectory}/coverage-reports, failed: $error") }
+                .ignoreElements()
+    }
+
     private fun runAndParseTests(chunkTimeoutSeconds: Long, testChunk: TestChunk): Single<List<AdbDeviceTestResult>> {
         val testPackageName = testChunk.testModuleInfo.moduleInfo.apkPackage.value
         val testRunnerClass = testChunk.testModuleInfo.testRunner.value
+        val coverageFileName = testChunk.testMethods.joinToString(",") { it.testName } + ".ec"
         val testMethodsArgs = "-e class " + testChunk.testMethods.joinToString(",") { it.testName }
         val timeout = Timeout(chunkTimeoutSeconds.toInt(), TimeUnit.SECONDS)
-
-        return processRunner.runAdb(
-                commandAndArgs = listOf(
-                        "-s", adbDevice.id,
-                        "shell", "am instrument -w -r $testMethodsArgs $testPackageName/$testRunnerClass"
-                ),
-                timeout = timeout)
+        val runCommand = if(args.testCoverageEnabled && args.pullTestFiles) {
+            "am instrument -w -r -e coverage true -e coverageFile ${args.testFilesPullDeviceDirectory}/coverage-reports/$coverageFileName $testMethodsArgs $testPackageName/$testRunnerClass"
+        } else {
+            "am instrument -w -r $testMethodsArgs $testPackageName/$testRunnerClass"
+        }
+        return processRunner.runAdb(commandAndArgs = listOf("-s", adbDevice.id, "shell", runCommand), timeout = timeout)
                 .ofType(Notification.Start::class.java)
                 .flatMap { instrumentationReader.readTestResults(it.output, chunkTimeoutSeconds) }
                 .doOnNext { instrumentationTestResult -> logTestResult(instrumentationTestResult) }
                 .map { instrumentationTestResult -> createAdbDeviceTestResult(instrumentationTestResult) }
                 .toList()
     }
 
+    private val Args.pullTestFiles get(): Boolean {
```

Comment:
Being picky here, but why not a standard method?
You're risking getting override by mistake for no benefits on my view.

Reply:
Style really. I'll replace it.

Reply:
👍